### PR TITLE
[WIP] ci: add test coverage threshold

### PR DESCRIPTION
**Pull request by KiranVarma02**

### CODE REVIEW

**Comment by alexluckett** [comment]
Defra uses Sonar as it's code quality tool, we can push this responsibility to Sonar instead of duplicating the check inside our repo.

It should be as simple as running a Sonar scan after the `npm run test` step in our `check-pull-request.yaml` pipeline and adding the property file.

Fancy giving it a go instead?

```yaml
      - name: SonarCloud Scan
        uses: SonarSource/sonarcloud-github-action@master
        env:
          SONAR_TOKEN: ${{ secrets. SONARCLOUD_TOKEN }}
```

And the updated properties:
https://docs.sonarsource.com/sonarcloud/enriching/test-coverage/javascript-typescript-test-coverage/#adding-the-coverage-analysis-parameter

Source: https://docs.sonarsource.com/sonarcloud/enriching/test-coverage/javascript-typescript-test-coverage/

File: .github/workflows/check-pull-request.yml [hunk 1]
```diff
@@ -161,3 +161,8 @@ jobs:
 
       - name: Run task
         run: ${{ matrix.task.run }}
+
+      - name: SonarCloud Scan
+        uses: SonarSource/sonarcloud-github-action@master
+        env:
+          SONAR_TOKEN: ${{ secrets.SONARCLOUD_TOKEN }}
```
Comment (KiranVarma02) [inline .github/workflows/check-pull-request.yml:168]: looks like this doesn't exist - @alexluckett

Comment (alexluckett) [reply .github/workflows/check-pull-request.yml:168]: Ah I do have one, I must not have applied it to the repo. I'll dig it out and apply it this afternoon.

**Comment by KiranVarma02** [comment]
not required

**Pull request closed without merging**